src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3410,8 +3410,14 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
             var memberSignatures = s_duplicateRecordMemberSignatureDictionary.Allocate();
             var membersSoFar = builder.GetNonTypeMembers(declaredMembersAndInitializers);
             var members = ArrayBuilder<Symbol>.GetInstance(membersSoFar.Count + 1);
+            var memberNames = PooledHashSet<string>.GetInstance();
             foreach (var member in membersSoFar)
             {
+                if (!memberNames.Contains(member.Name))
+                {
+                    memberNames.Add(member.Name);
+                }
```

Comment:
Doesn't `Add` method internally check whether the element is there or not?

Reply:
It does. Thanks

Reply:
The behavior of Add is to throw when the key is already present. Is that what we want? Consider testing a scenario where two fields with the same name are declared in the type.

Reply:
@RikkiGibson Isn't it supposed to return `false` when the item already exists?

Reply:
Oops, sounds like I mixed up the behavior for Dictionary vs for Set. 😅

Reply:
I did the same :-P Had to check the docs ;-)